### PR TITLE
Update App to React 15 and latest React-Bootstrap

### PR DESCRIPTION
**Pull request by mukundgoel**
Was playing around with this app and realized it was using an older version of `React` and `react-bootstrap`.

This is an attempt to upgrade the app to the latest version of `React` and `react-bootstrap`.

Screenshot after upgrade:
image

/cc @ryan-roemer

### CODE REVIEW

File: templates/index.jsx [hunk 1]
```diff
@@ -26,14 +26,13 @@ export default class Index extends React.Component {
           <meta name="apple-mobile-web-app-capable" content="yes" />
 
           {/* Polyfills before CSS */}
-          <meta name="ie-shiv-polyfill" dangerouslySetInnerHTML={{__html:
+          <span dangerouslySetInnerHTML={{__html:
```
Comment (mukundgoel) [inline templates/index.jsx:29]: The new version of react does not like using `dangerouslySetInnerHTML` inside a `meta` tag. Is there a better alternative to this? Maybe doing `<script src="${CLOUDFLARE}/html5shiv/3.7.2/html5shiv-printshiv.js"></script>` without a `meta` tag?

File: test/client/spec/components/types-title.spec.jsx [hunk 2]
```diff
@@ -1,48 +0,0 @@
-/**
```
Comment (ryan-roemer) [inline test/client/spec/components/types-title.spec.jsx:1]: This is our project for _teaching_ testing in addition to app development, so we really can't remove the tests....

Comment (ryan-roemer) [inline test/client/spec/components/types-title.spec.jsx:1]: But the tests are fine being updated to modern plugins or enzyme.

Comment (mukundgoel) [inline test/client/spec/components/types-title.spec.jsx:1]: Sure I will add them back.

**Comment by mukundgoel** [comment]
hey @ryan-roemer appreciate the re-review.. Fixed the tests to incorporate the new ReactTestUtils from Facebook. Please let me know if anything else is needed. Thanks!

**Comment by mukundgoel** [comment]
So the appveyor CI failed apparently but the other tasks passed... everything is passing on local as well.

```
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.12
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "node" "C:\\projects\\converter-react\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v0.12.13
npm ERR! npm  v3.9.3

npm ERR! Callback called more than once.
npm ERR! 
npm ERR! If you need help, you may report this error at:
```

should i re-trigger?

I guess https://github.com/strongloop/fsevents/issues/132 might be related

```
npm ls fsevents
converter-react@0.0.1
└─┬ babel@5.8.38
  └─┬ chokidar@1.5.0
    └── fsevents@1.0.12 
```

**Comment by mukundgoel** [comment]
🎉 thanks for helping out with CI 👍

**Comment by ryan-roemer** [comment]
Finally digging in here!

Seeing this warning in functional tests:

```
Warning: Failed propType: Invalid prop `children` supplied to `Provider`, expected a single ReactElement.
With React 0.14 and later versions, you no longer need to wrap <Provider> child into a function.
Warning: Failed propType: The prop 'id' is required to make 'Dropdown' accessible for users using assistive technologies such as screen readers Check the render method of `Uncontrolled(Dropdown)`.
Warning: validateDOMNesting(...): <span> cannot appear as a child of <head>. See Index > head > span.
```

Is this something we need to fix?

**Comment by ryan-roemer** [comment]
Opening issue of https://github.com/FormidableLabs/converter-react/issues/49 instead of fixing here. Thanks @mukundgoel !

**Pull request merged**